### ISSUE TITLE
Feature Request

### ISSUE THREAD

**Issue by jocelynduc** [RailsApps/rails3-bootstrap-devise-cancan#50]
Hello,

First of all, thank you for your great tutorials and example applications.

I am looking for a new example using devise, cancan and doorkeeper or other OAuth implementation.

In other words, here are my needs:
1. A main application (provider) with some functionalities with user authentification and autorization with devise and cancan, such as your rails3-bootstrap-devise-cancan example application. That main application will have an API (REST interface)  that will be used by some RoR and/or PHP clients. My goal is to secure the access to that API with a token system as OAuth
2. A RoR and/or PHP client that will used the provider's API after his authentication.

I hope you have understood my request and I am sure it will be useful for others.

Feel free to contact me for any other information.

Jocelyn Duc

**Comment by DanielKehoe** [comment]
I've been thinking about creating an example app showing how to use Rails as a backend for an iOS app. Would your requirements be the same?

**Comment by jocelynduc** [comment]
Yes, I have been thinking about consuming data from a Titanium mobile app too. So, yes, a such example will help me.

The important points, in my case, are:
1. One unique RoR website as provider that will have a frontend and a backend (users / admins)
2. Several types of client types: RoR websites, PHP websites and mobile apps
3. Client applications will have to be authenticated to the provider to use provider's REST interface.
4. Regular users will be able to log on the provider.and on client apps.

Another thing I need for my project is to do some redirection such as PayPal system, for example:
1. User starts a processus on a known and registered client website
2. User is so redirected on my provider RoR website
3. User will have to complete his processus on the provider RoR website (forms, etc..)
4. Then, user will be redirected on the source client website.
